src/v/pandaproxy/schema_registry/service.cc

```diff
@@ -227,17 +245,29 @@ ss::future<> service::mitigate_error(std::exception_ptr eptr) {
         return ss::now();
     }
     vlog(plog.warn, "mitigate_error: {}", eptr);
-    return ss::make_exception_future<>(eptr).handle_exception_type(
-      [this, eptr](kafka::client::broker_error const& ex) {
+    return ss::make_exception_future<>(eptr)
+      .handle_exception_type(
+        [this, eptr](kafka::client::broker_error const& ex) {
+            if (
+              ex.error == kafka::error_code::sasl_authentication_failed
+              && _has_ephemeral_credentials) {
+                return inform(ex.node_id).then([this]() {
+                    // This fully mitigates, don't rethrow.
+                    return _client.local().connect();
+                });
+            }
+
+            // Rethrow unhandled exceptions
+            return ss::make_exception_future<>(eptr);
+        })
+      .handle_exception_type([this,
+                              eptr](kafka::client::topic_error const& ex) {
           if (
-            ex.error == kafka::error_code::sasl_authentication_failed
+            ex.error == kafka::error_code::topic_authorization_failed
```

Comment:
note: I guess this could be a result of creating, reading, or writing a topic. The latter two failure modes would probably require the created ACLs to have been modified, which is probably quite tricky. I'm not sure it's worth handling that.